odl/space/cu_ntuples.py

```diff
@@ -847,36 +976,271 @@ def _norm_default(x):
     return x.data.norm()
 
 
+def _pnorm_default(x, p):
+    if p == float('inf'):
+        raise NotImplementedError('inf-norm not implemented.')
+    elif int(p) != p:
+        raise NotImplementedError('non-integer powers not implemented')
+    # TODO: optimized version in C++ code?
+    xp = abs(x)
+    xp **= p
+    return sum(xp)**(1/p)
```

Comment:
[Not big OOPS after all](https://github.com/odlgroup/odl/blob/issue-246__l2discr/odl/space/cu_ntuples.py#L967-L968). `sum` is also shadowed, that's why I have to [re-import the built-in one](https://github.com/odlgroup/odl/blob/issue-246__l2discr/odl/space/cu_ntuples.py#L543-L544) at one place.